episodes/anonymization.md

```diff
@@ -32,19 +33,352 @@ Metadata elements in imaging, such as patient names and addresses, are often cle
 
-However, a key issue under current investigation is that some defacing algorithms may inadvertently alter more than just the facial features. Emerging research, including studies still in pre-print, suggests that these algorithms might also affect the morphometry of the brain image. This could lead to the unintended loss or distortion of critical data. Therefore, it is advisable to proceed with caution and, whenever possible, compare the original and defaced images to ensure that important information remains intact and unaltered.
 
-![Image from "A reproducibility evaluation of the effects of MRI defacing on brain segmentation" by Chenyu Gao, Bennett A. Landman, Jerry L. Prince, and Aaron Carass. The preprint is available [here](https://pubmed.ncbi.nlm.nih.gov/37293070/).](fig/deface-example.jpg){alt='Defacing examples'}
+We could in theory write our own defacing algorithm. For such an algorithm either SITK or skimage provide several useful built in functions including [morphological operations](learners/reference.md#Morphological operations) such as erosion, dilation, grow-from-seed and other types of operations such as connected component analyses and masking. 
 
-### Text on Images
+::::::::::::::::::::::::::::::::::::::: challenge 
+
+## Challenge: Soft tissue stripping(Optional)
+
+Look at the head MRI above. Try using SITK to get rid off some of the soft tissue in the image (already loaded into sag_image). 
+
+
+::::::::::::::: solution
+
+## Solution
+
+We can apprach the problem in various ways. Below is one solution with erosion, dilation and masking:
+
+```python
+# Apply thresholding to remove soft tissues by first taking out the air then dilating and cleaning
+# the assumption is that the black around the brain is zero and low values
+# Create the brain mask
+lower_thresh = 0
+upper_thresh = 100
+brain_mask = sitk.BinaryThreshold(sag_image, lowerThreshold=lower_thresh, upperThreshold=upper_thresh)
+
+# Morphological operations to clean the mask
+brain_mask_cleaned = sitk.BinaryDilate(brain_mask, [5, 5, 5])
+brain_mask_cleaned = sitk.BinaryErode(brain_mask_cleaned, [5, 5, 5])
+
+# Display the original and cleaned mask images using the general display function
+interact(
+    display_images,
+    image1_z=(0, brain_mask.GetSize()[2] - 1),
+    image2_z=(0, brain_mask_cleaned.GetSize()[2] - 1),
+    image1_npa=fixed(sitk.GetArrayViewFromImage(brain_mask)),
+    image2_npa=fixed(sitk.GetArrayViewFromImage(brain_mask_cleaned)),
+    title1=fixed("Original Mask"),
+    title2=fixed("Cleaned Mask")
+)
+```
+Then we can further clean with a connection component analysis that throws out small floaters, and use the inverse as out final mask.
+
+```python
+
+def keep_largest_component(mask_image):
+    # Ensure mask_image is binary (0 and 1 values)
+    mask_image = sitk.Cast(mask_image, sitk.sitkUInt8)
```

Comment:
I don't understand this, uint8 has a range of 0..255, so how does this guarantee 0..1?